addons/pos_restaurant/static/src/js/Screens/TicketScreen.js

```diff
@@ -73,6 +73,13 @@ odoo.define('pos_restaurant.TicketScreen', function (require) {
             }
             async setTip(order, serverId, amount) {
                 try {
+                    const paymentline = order.get_paymentlines()[0];
+                    if (paymentline.payment_method.payment_terminal) {
+                        paymentline.amount += amount;
+                        this.env.pos.set_order(order, {silent: true});
```

Comment:
Calling set_order doesn't seem to be necessary.

Reply:
We only give the `cid` of the payment line to `send_payment_adjust` so the correct order should be set to be able to find it.
I first thought it would be a better option to give the order as a second argument but the signature of this function would have been different from all other functions of the payment interface.